### PR TITLE
Fix: Validate against the byte length of strings, rather than their unicode length

### PR DESCRIPTION
**Pull request by LordAro**
See also OpenTTD/OpenTTD#6389

### CODE REVIEW

File: bananas_api/helpers/api_schema.py [hunk 1]
```diff
@@ -77,12 +82,12 @@ class Global(OrderedSchema):
     read_only = ["archived", "replaced_by"]
 
     # Most of these limits are limitations in the OpenTTD client.
-    name = fields.String(validate=validate.Length(max=31))
+    name = fields.String(validate=validate.BytesLength(max=31))
```
Comment (TrueBrain) [inline bananas_api/helpers/api_schema.py:85]: I am 100% sure you did not test this. don't be lazy, start a local setup, and make sure this works please. README is hopefully pretty clear, and via docker it is really easy. Please don't be "that guy" :D :D

**Comment by LordAro** [comment]
Suitably tested?

File: bananas_api/helpers/api_schema.py [hunk 2]
```diff
@@ -64,6 +64,18 @@ def __call__(self, value):
         return super().__call__(value)
 
 
+class ValidateBytesLength(validate.Length):
+    def __call__(self, value):
+        super().__call__(value.encode())
+
+    def _format_error(self, value, message):
+        # There were no issues encoding it in __call__,
```
Comment (TrueBrain) [inline bananas_api/helpers/api_schema.py:72]: personally, I would just do: `value = value.decode()`, and call the parent.

Given you did `encode()` without parameter, no need to do `decode()` with parameter.

File: bananas_api/helpers/api_schema.py [hunk 3]
```diff
@@ -64,6 +64,18 @@ def __call__(self, value):
         return super().__call__(value)
 
 
+class ValidateBytesLength(validate.Length):
+    def __call__(self, value):
+        super().__call__(value.encode())
```
Comment (TrueBrain) [inline bananas_api/helpers/api_schema.py:69]: missing `return` and `decode()`

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**